plugins/PrivacyManager/PrivacyManager.php

```diff
@@ -582,4 +586,22 @@ private static function getMaxGoalId()
     {
         return Db::fetchOne("SELECT MAX(idgoal) FROM " . Common::prefixTable('goal'));
     }
+
+    /**
+     * Returns a unique salt used for pseudonimisation of user id only
+     *
+     * @return string
+     */
+    public static function getUserIdSalt()
+    {
+        static $salt = null;
+        if (is_null($salt)) {
```

Comment:
Is this cache really needed? The `Option` class would cache it anyway? Such a static property makes it bit complicated for tests.